test/models/deposit_test.rb

```diff
@@ -0,0 +1,59 @@
+require 'test_helper'
+
+class DepositTest < ActiveSupport::TestCase
+  def setup
+    super
+
+    @deposit = Deposit.new(registrar: registrars(:bestnames))
+    @minimum_deposit = Setting.minimum_deposit
+    Setting.minimum_deposit = 1.00
+  end
+
+  def teardown
+    super
+
+    Setting.minimum_deposit = @minimum_deposit
+  end
+
+  def test_validate_amount_cannot_be_lower_than_0_01
+    Setting.minimum_deposit = 0.0
+    @deposit.amount = -10
+    refute(@deposit.valid?)
+    assert(@deposit.errors.full_messages.include?("Amount is too small. Minimum deposit is 0.01 EUR"))
+  end
+
+  def test_validate_amount_cannot_be_lower_than_minimum_deposit
+    @deposit.amount = 0.10
+    refute(@deposit.valid?)
+
+    assert(@deposit.errors.full_messages.include?("Amount is too small. Minimum deposit is 1.0 EUR"))
+  end
+
+  def test_registrar_must_be_set
```

Comment:
How about approach used by Thoughtbot, when you omit association validation in favor of DB constraint, given it's anyway cannot be fixed by a user?

https://robots.thoughtbot.com/validation-database-constraint-or-both

Reply:
This is not a database table, it's a class that uses ActiveRecord validations. A form object, if you'd like.

Reply:
hm, ok. Missed that fact.